mmf/datasets/builders/glue/builder.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import torch
+from mmf.common.registry import registry
+from mmf.common.sample import Sample
+from mmf.datasets.base_dataset import BaseDataset
+from mmf.datasets.mmf_dataset_builder import MMFDatasetBuilder
+from mmf.utils.general import retry_n
+
+from datasets import load_dataset
+
+
+MAX_RETRIES = 9
+
+
+class GLUEDataset(BaseDataset):
+    DATASET_KEY_MAP = {
+        "text_a": {
+            "glue_mnli_mismatched": "premise",
+            "glue_qnli": "question",
+            "glue_sst2": "sentence",
+            "glue_qqp": "question1",
+        },
+        "text_b": {
+            "glue_mnli_mismatched": "hypothesis",
+            "glue_qnli": "sentence",
+            "glue_sst2": None,
+            "glue_qqp": "question2",
+        },
+    }
+
+    def __init__(self, config, dataset_type, imdb_idx):
+        dataset_name = f"glue_{config.task}"
+        super().__init__(dataset_name, config, dataset_type, imdb_idx)
+
+        if dataset_type == "val":
+            # datasets library uses validation as val set.
+            dataset_type = "validation"
+
+        # For MNLI-MM, set train to MNLI-M train
+        task = config.task
+        if config.task.startswith("mnli") and dataset_type == "train":
+            task = "mnli"
+
+        self.dataset = retry_n(
+            MAX_RETRIES, load_dataset, "glue", task, split=dataset_type
+        )
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __getitem__(self, idx):
+        annotation = self.dataset[idx]
+        current_sample = Sample()
+        text_processor_input = {
+            "text_a": annotation[self.DATASET_KEY_MAP["text_a"][self.dataset_name]]
+        }
+
+        text_b = annotation.get(self.DATASET_KEY_MAP["text_b"][self.dataset_name], None)
```

Comment:
I added a fix-up commit for the case of `text_b` non-existence (this line is the new version). @apsdehal Could you import it again? Thanks 😃